src/backend/joanie/core/serializers.py

```diff
@@ -1,28 +1,193 @@
+        ).data
+
+        if user is not None:
+            for index, course_run in enumerate(course_runs):
+                try:
+                    enrollment = models.Enrollment.objects.get(
+                        course_run__pk=course_run["id"],
+                        user__username=user,
+                        order__product=product,
+                    )
+                    course_runs[index] = CourseRunEnrollmentSerializer(enrollment).data
+
+                except ObjectDoesNotExist:
+                    pass
+
+        return course_runs
 
 
 class ProductSerializer(serializers.ModelSerializer):
     """
-    Product serializer including list of course runs with its positions
+    Product serializer including
+        - certificate information if there is
+        - targeted courses with its course runs
+            - If user is authenticated, we try to retrieve enrollment related
+              to each course runs.
+        - order if user is authenticated
     """
 
     id = serializers.CharField(source="uid", read_only=True)
+    certificate = CertificationDefinitionSerializer(
+        read_only=True, source="certificate_definition"
+    )
+    currency = serializers.SerializerMethodField()
+    order = serializers.SerializerMethodField(read_only=True)
+    price = serializers.DecimalField(
+        max_digits=9,
+        decimal_places=2,
+        coerce_to_string=False,
+        min_value=0,
+    )
+    target_courses = serializers.SerializerMethodField()
 
     class Meta:
         model = models.Product
-        fields = ["id", "title", "call_to_action", "price"]
+        fields = [
+            "call_to_action",
+            "certificate",
+            "currency",
+            "id",
+            "order",
+            "price",
+            "target_courses",
+            "title",
+            "type",
+        ]
+
+    def get_target_courses(self, product):
+        """
+        For the current product, retrieve its related courses.
+        """
+
+        context = self.context.copy()
+        context.update({"product": product})
+
+        return TargetCourseSerializer(
+            instance=models.Course.objects.filter(
+                product_relations__product=product
+            ).order_by("product_relations__position"),
+            many=True,
+            context=context,
+        ).data
+
+    def get_currency(self, _):  # pylint: disable=no-self-use
+        """
+        Return currency code and symbol
+        to allow frontend to format properly price
+        """
+        (code, symbol) = settings.JOANIE_CURRENCY[:2]
+        return {"code": code, "symbol": symbol}
+
+    def get_order(self, product):
+        """
+        For the current product, retrieve valid order related
+        to the authenticated user.
+        """
+        user = self.context.get("user", None)
+        order = None
+
+        if user is not None:
+            try:
+                query = models.Order.objects.get(
```

Comment:
actually this is not returning a query but an instance of order.
Then you could rename your "order" variable to "order_data"?
```suggestion
                order = models.Order.objects.get(
```

Reply:
Good suggestion, thanks ! 🙏

Reply:
you forgot to push this change?